flask/namespaces.py

```diff
@@ -1,8 +1,95 @@
+import re
 from app import app
 from mongo import db
 from flask import request, jsonify
 from app import swagger
 from flasgger.utils import swag_from
+from packages import checkUserUnauthorized
+
+from datetime import datetime
+from auth import generate_uuid
+
+@app.route("/namespaces", methods=["POST"])
+def create_namespace():
+    uuid = request.form.get("uuid")
+
+    if not uuid:
+        return jsonify({"code": 401, "message": "Unauthorized"}), 401
+    
+    # Get the user document from the uuid.
+    user_doc = db.users.find_one({"uuid": uuid})
+
+    if not user_doc:
+        return jsonify({"code":  401, "message": "Unauthorized"}), 401
+    
+    namespace_name = request.form.get("namespace")
+
+    if not namespace_name:
+        return jsonify({"code": 400, "message": "Please enter namespace name"}), 400
+    
+    pattern = r'^[a-zA-Z0-9_-]+$'
+
+    # Make sure namespace name only contains [a-z], [A-Z], [0-9], - and _ characters.
+    if not re.match(pattern, namespace_name):
+        return jsonify({"code": 400, "message": "Namespace name can only include (a-z), (A-Z), (0-9), - and _"}), 400
+
+    # Get the namespace document from the namespace name.
+    # To check if already a namespace exists by this name.
+    namespace_doc = db.namespaces.find_one({"namespace": namespace_name})
+
+    # Check if namespace already exists.
+    if namespace_doc:
+        return jsonify({"code": 400, "message": "Namespace already exists"}), 400
+
+    namespace_obj = {
+        "namespace": namespace_name,
+        "createdAt": datetime.utcnow(),
+        "author": user_doc["_id"],
+        "maintainers": [user_doc["_id"]],
+        "admins": [user_doc["_id"]]
+    }
+
+    db.namespaces.insert_one(namespace_obj)
+
+    return jsonify({"code": 200, "message": "Namespace created successfully"}), 200
+
+@app.route("/namespaces/<namespace_name>/uploadToken", methods=["POST"])
+def create_upload_token(namespace_name):
+    uuid = request.form.get("uuid")
+
+    if not uuid:
+        return jsonify({"code": 401, "message": "Unauthorized"}), 401
+    
+    user_doc = db.users.find_one({"uuid": uuid})
+
+    if not user_doc:
+        return jsonify({"code": 401, "message": "Unauthorized"}), 401
+    
```

Comment:
Again, this can be bypassed by knowing the uuid and sending it in a mock form field. I'm no expert in this but I feel like authentication should be dealt differently.